binaries/daemon/src/lib.rs

```diff
@@ -275,6 +278,31 @@ impl Daemon {
                 }
                 (None, RunStatus::Continue)
             }
+            DaemonCoordinatorEvent::Logs {
+                dataflow_id,
+                node_id,
+            } => {
+                // read file
+                let logs = async {
+                    let log_dir = temp_dir();
+
+                    let mut file = File::open(log_dir.join(format!("{dataflow_id}-{node_id}.txt")))
+                        .await
+                        .wrap_err("Could not open log file")?;
+
+                    let mut contents = vec![];
+                    file.read_to_end(&mut contents)
+                        .await
+                        .wrap_err("Could not read content of log file")?;
```

Comment:
This read could take quite some time if the file is large. So ideally, we would launch this operation in a background task using `tokio::spawn` instead of blocking the main loop of the daemon (which will delay other events). This will probably require some refactoring of the event handling code, so I can look into it in a follow-up PR if you prefer.

Reply:
So, I have refactored the `daemon handle_coordinator_event` by transfering the ownership of the `reply_tx` channel to the different `event` instead of reporting a reply for all `event` to a single `reply_tx`. 

This makes it possible to spawn a background task as you said and take ownership of the `reply_tx`.